Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -370,6 +370,37 @@ public extension Tensor {
             rangeFrom: 0, to: Int32(rank), stride: 1)
         return transposed(permutation: Tensor<Int32>(defaultPermutations))
     }
+    
+    /// Returns a tensor with specified dimensions reversed.
+    /// - Precondition: Each value in `axes` must be in the range `-rank..<rank`.
+    /// - Precondition: There must be no duplication in `axes`.
+    @inlinable
+    @differentiable(wrt: self where Scalar: TensorFlowFloatingPoint)
+    func reversed(axes: Tensor<Int32>) -> Tensor {
```

Comment:
Yes, one type. Ideally something that is possible to add some sort of small vector optimization to (vs an actual Set or something). I personally think Int... are more trouble than they're worth as well. We can move there more easily by adding overloads to raw ops that that take [Int] and then call into the Tensor<Int32> versions. That way, no public apis would have Tensor<Int32> versions. I think we would probably need to convert and remove all of the dataset apis first because this would make tracing those functions probably not work.

Reply:
Note: Ed Connell has been working on a fast SIMD-backed type for Tensor shapes in SwiftRT and that appears to be working very well.

Reply:
> Note: Ed Connell has been working on a fast SIMD-backed type for Tensor shapes in SwiftRT and that appears to be working very well.

That's cool! I see `TensorShape` and `AxisSet` as distinct types, but perhaps they can share some logic.

Reply:
They absolutely should be separate types; just a pointer / inspiration for implementation.